sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/TimestampFormatter.scala

```diff
@@ -71,6 +71,19 @@ sealed trait TimestampFormatter extends Serializable {
       s"The method `parseWithoutTimeZone(s: String)` should be implemented in the formatter " +
         "of timestamp without time zone")
 
+  /**
+   * Returns true if the parsed timestamp contains the time zone component, false otherwise.
+   * Used to determine if the timestamp can be inferred as timestamp without time zone.
+   *
+   * @param s - string with timestamp to inspect
+   * @return whether the timestamp string has the time zone component defined.
+   */
+  @throws(classOf[IllegalStateException])
+  def isTimeZoneSet(s: String): Boolean =
```

Comment:
This API looks a bit weird. How about we throw exception in `parseWithoutTimeZone` if the input string has timezone?

BTW is it possible to statically fail the ntz formatter creating if the pattern string contains timezone? Then we can fail earlier before processing data.

Reply:
I can update.

IMHO, it is tricky to check this statically as there does not seem to be a way of checking pattern components and we need an actual value to validate whether or not it has a zone offset.